types/index.d.ts

```diff
@@ -337,5 +337,13 @@ declare namespace jest {
      * @param {Array.<String>} substring
      */
     toIncludeMultiple(substring: string[]): R;
+
+    /**
+     * Use `.toThrowWithMessage` when checking if a callback function throws an error of a given type with a given error message.
+     *
+     * @param {Function} type
+     * @param {String | RegExp} message
+     */
+    toThrowWithMessage(type: Function, message: string | RegExp): R;
```

Comment:
This needs adding to the `expect` interface too

Reply:
Ah we just merged the types for asymmetric matchers so needed to update this file twice - all good though I just did it for you :)